keel-core/src/main/kotlin/com/netflix/spinnaker/keel/persistence/ArtifactRepository.kt

```diff
@@ -11,10 +11,11 @@ interface ArtifactRepository {
   /**
    * Creates or updates a registered artifact
    */
-  // todo eb: https://github.com/spinnaker/keel/issues/655
   fun register(artifact: DeliveryArtifact)
 
-  fun get(name: String, type: ArtifactType): DeliveryArtifact
+  fun get(name: String, type: ArtifactType, deliveryConfigName: String): List<DeliveryArtifact>
```

Comment:
What's the rationale for now returning a list? This should be unique, surely?

Reply:
You might have two artifacts of the same name and type, but different sorting or release status. For example, you may care about only a snapshot of a deb for a feature branch env but for the main environment you care about release status.

I think I'd model that as two artifacts:
```
artifacts:
  - name: keel
     type: deb
     statuses: 
     - snapshot
     reference: feature-stack-artifact
  - name: keel
     type: deb
     statuses: 
     - release
     reference: main-artifact
```